### PR TITLE
Optionally pass cortex config as configmap.

### LINKED ISSUE #235: Deploy the config as a `configMap` rather than a secret
Let's also add the ability to deploy the config as a `configMap` rather than a secret?

_Originally posted by @cabrinha in https://github.com/cortexproject/cortex-helm-chart/issues/227#issuecomment-930516408_

And have the credentials passed as environment variables backed by secrets? That would be quite optimal.

### PR DESCRIPTION
**Pull request by jmcarp**
**What this PR does**:
Optionally pass cortex config as configmap.

Note: I only updated the ingestor templates so far, to verify that the approach makes sense. If it does, I'll copy and paste the updated volume to the other templates.

**Which issue(s) this PR fixes**:
Fixes #235.

**Checklist**
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by ShuzZzle** [comment]
Nice!

with that approach however, we should really have the s3 credentials (or similiar) **not** in the configMap

File: values.yaml [hunk 1]
```diff
@@ -34,6 +34,7 @@ serviceAccount:
   annotations: {}
   automountServiceAccountToken: true
 
+useConfigMap: true
```
Comment (ShuzZzle) [inline values.yaml:37]: I wouldn't enable this per default personally

**Review by ShuzZzle — CHANGES_REQUESTED** [review]
Hey, thank you for your contribution.

Can you please add the CHANGELOG.md entry?
Also we would need to add a mechanism similiar to: [This](https://github.com/cortexproject/cortex-helm-chart/blob/master/templates/ingester/ingester-dep.yaml#L33) 

So we can restart components when configmap changes, right now we can't use something like configmap reloader AFAIK

**Comment by jmcarp** [comment]
Updated, lmk if I missed anything.

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@
 * [CHANGE] Disable ingester liveness probes by default. #263
 * [BUGFIX] workaround for .Capabilities.APIVersions. #277
 * [DEPENDENCY] Update Helm release memcached to v5.15.9 #273
+* [FEATURE] Optionally manage cortex config as configmap. #280
```
Comment (ShuzZzle) [inline CHANGELOG.md:12]: technically FEATURE's should be the first entry in the changelog

**Review by ShuzZzle — COMMENTED** [review]
I will have another glance tomorrow but this looks really good. Thanks!

**Review by ShuzZzle — APPROVED** [review]
I re-edited the guide, because inline code blocks completely broke the the text unfortunately. We will have to adjust the markdown processor something like that in the future

File: templates/configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{{- if (and (not .Values.useExternalConfig) (.Values.useConfigMap)) }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "cortex.fullname" . }}
```
Comment (ShuzZzle) [inline templates/configmap.yaml:5]:
```suggestion
  name: {{ template "cortex.fullname" . }}-config
```
tiny improvement idea

**Comment by ShuzZzle** [comment]
@jmcarp Please be careful when rebasing, my changes to the guide were lost

**Review by kd7lxl — APPROVED** [review]

**Comment by jmcarp** [comment]
Ack, sorry, not used to collaborating on a branch 😓

**Pull request merged**